twake/backend/node/src/core/platform/services/database/services/connectors/cassandra.ts

```diff
@@ -41,11 +63,57 @@ export class CassandraConnector extends AbstractConnector<
     );
   }
 
+  async isKeyspaceCreated(): Promise<boolean> {
+    let result;
+
+    try {
+      result = await this.client.execute(
+        `SELECT * FROM system_schema.keyspaces where keyspace_name = '${this.options.keyspace}'`,
+      );
+
+      if (!result) {
+        throw new Error("No result for keyspace query");
+      }
+    } catch (err) {
+      throw new Error("Keyspace query error");
+    }
+
+    return result ? Promise.resolve(true) : Promise.reject(new Error("Keyspace not found"));
+  }
+
+  waitForKeyspace(delay: number = 500, retries: number = 10): Promise<boolean> {
+    const subject = new Subject<boolean>();
+    const obs$ = defer(() => this.isKeyspaceCreated());
+
+    obs$
+      .pipe(
+        retryWhen(errors =>
+          errors.pipe(
+            delayWhen((_, i) => timer(i * delay)),
+            //delay(1000), if we want fixed delay
+            tap(() => logger.debug("Retrying...")),
+            take(retries),
+            concat(throwError("Maximum number of retries reached")),
+          ),
+        ),
+      )
+      .subscribe(
+        () => logger.debug("Keyspace has been found"),
+        err => {
+          logger.error({ err }, "Error while getting keyspace information");
+          subject.error(new Error("Can not find keyspace information"));
+        },
+        () => subject.complete(),
+      );
+
+    return subject.toPromise();
+  }
+
   async drop(): Promise<this> {
     try {
```

Comment:
Do we really want to have this in our code ? We will never automate keyspace deletion, it seems dangerous to me to have this somewhere 💥